lib/whitehall_importer/edition_history.rb

```diff
@@ -64,6 +75,31 @@ def editors
       revision_history.pluck("whodunnit").uniq
     end
 
+    def timeline_entry(event, edition_number)
+      if event["event"] == "create" && edition_number == 1
+        "first_created"
+      elsif event["event"] == "create" && edition_number > 1
+        "new_edition"
+      elsif previous_event(event) && previous_event(event)["state"] == event["state"]
+        "document_updated"
+      else
+        whitehall_state_mapping = {
+          "archived" => "archived",
+          "draft" => "removed",
+          "published" => "published",
+          "rejected" => "rejected",
+          "scheduled" => "scheduled",
+          "submitted" => "submitted",
+          "withdrawn" => "withdrawn",
+        }
+        begin
+          whitehall_state_mapping.fetch(event["state"])
+        rescue KeyError
+          raise AbortImportError, "Edition history has an unsupported state"
```

Comment:
I don't know if there's some cleaner away of achieving a raise of `AbortImportError` when the key doesn't exist?